Assets/Scripts/PlayScene/PlayScene.cs

```diff
@@ -54,8 +54,58 @@ private IEnumerator PlayerDeath()
         player.gameObject.SetActive(true);
         running = true;
     }
-    
 
-  
+    #region BlockHelperFunctions
 
+    public void RespawnRespawnBlocks(RespawnBlock respawnBlock, float respawnTime)
```

Comment:
Should in my opinion be handled within the respawnBlock

Reply:
The coroutine stops if you put the code within the respawnBlock and I don't think just disabling the collider is a better idea...